### PR TITLE
Add org.freeorion.FreeOrion

### PR DESCRIPTION
**Pull request by Siosm**
Initial import with 0.4.10 release.

Upstream has been contacted and is OK with me handling Flathub packaging: https://github.com/freeorion/freeorion/issues/1914

This will probably require some iterations to get all permissions/access right.

### CODE REVIEW

File: appdata.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
+diff --git a/packaging/org.freeorion.FreeOrion.metainfo.xml b/packaging/org.freeorion.FreeOrion.metainfo.xml
```
Comment (barthalion) [inline appdata.patch:1]: Has it been upstreamed?

File: appdata.patch [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
++    <release version="0.4.6" date="2016-09-22"/>
++    <release version="0.4.5" date="2015-09-08"/>
++    <release version="0.4.4" date="2014-09-07"/>
++    <release version="0.4.3" date="2013-10-26"/>
++    <release version="0.4.2" date="2013-02-20"/>
++    <release version="0.4.1" date="2012-08-03"/>
++    <release version="0.4" date="2012-02-05"/>
++  </releases>
++  <content_rating type="oars-1.1">
++    <content_attribute id="social-chat">intense</content_attribute>
```
Comment (barthalion) [inline appdata.patch:24]: Is it accurate? I believe there's some violence depicted in the game.

File: org.freeorion.FreeOrion.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+runtime-version: "19.08"
+sdk: org.freedesktop.Sdk
+
+finish-args:
+  - --device=dri
+  - --share=ipc
+  - --share=network
+  - --socket=pulseaudio
+  - --socket=wayland
+  - --socket=x11
```
Comment (barthalion) [inline org.freeorion.FreeOrion.yaml:14]: Should be fallback-x11.

File: org.freeorion.FreeOrion.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  - shared-modules/glew/glew.json
+  - shared-modules/glu/glu-9.json
+
+  - name: boost
+    buildsystem: simple
+    build-commands:
+      - ./bootstrap.sh --prefix=${FLATPAK_DEST} --with-python=/usr/bin/python3
+           --with-libraries=filesystem,iostreams,locale,log,serialization,system,thread,python
+      - ./b2 --reconfigure -j ${FLATPAK_BUILDER_N_JOBS} toolset=gcc threading=multi
+          variant=release debug-symbols=off link=shared,static runtime-link=shared
```
Comment (barthalion) [inline org.freeorion.FreeOrion.yaml:26]: That seems more complicated than existing boost builds. Is there any reason to include both shared and static libraries?

File: org.freeorion.FreeOrion.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+finish-args:
+  - --device=dri
+  - --share=ipc
+  - --share=network
+  - --socket=pulseaudio
+  - --socket=wayland
+  - --socket=x11
+
+modules:
```
Comment (barthalion) [inline org.freeorion.FreeOrion.yaml:16]: I'm sure there are at least static libraries and development headers to clean up.

File: appdata.patch (hunk 1 above)
Comment (Siosm) [reply appdata.patch:1]: https://github.com/freeorion/freeorion/pull/3146 & https://github.com/freeorion/freeorion/pull/3145

**Comment by nedrichards** [comment]
bot, build org.freeorion.FreeOrion

File: org.freeorion.FreeOrion.yaml (hunk 4 above)
Comment (Siosm) [reply org.freeorion.FreeOrion.yaml]: Removed static libs.

Comment (Siosm) [reply org.freeorion.FreeOrion.yaml]: The other libraries are explicitly required for the game.

**Comment by Siosm** [comment]
Thanks for the review. The latest push should have fixed most of the issue. Sound is working fine for me.

**Comment by barthalion** [comment]
Looks good.

@Vezzra, would you like to be invited to the Flathub repository? Just in case you want to do something yourself.

**Comment by Vezzra** [comment]
@barthalion, thanks for the offer, but since I don't know anything about creating or maintaining FlatPak packages, I wouldn't know what I'm doing. And unfortunately I don't have the time to dig into that stuff. I can just barely keep up with my current involvement with the FreeOrion project and my responsibilities there, I've no capacities left for more.

When it comes to providing and maintaining distribution channels like AppImage, Snap or FlatPak packages, this has to be done by other volunteers willing to take over these tasks.

In light of all this it does not make much sense to grant me access to the Flathub repo. But if there should ever be a change in the state of things and I start to have time for maintaining FlatPak packages etc. for FreeOrion, I'll get back to your offer. 😃

**Comment by barthalion** [comment]
@Vezzra Understood! It's available at https://flathub.org/apps/details/org.freeorion.FreeOrion now, in case you want to link to it on the project's website.

**Pull request closed without merging**